### PR TITLE
[AKS] Fix the bug related to AKS Monitoring MSI auth when the location value with spaces

### PR DESCRIPTION
**Pull request by ganga1980**
**Related command**

**Description**

This PR fixes the bug related to AKS Monitoring MSI auth when the location  value of the workspace GET response has spaces in it.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by ganga1980** [comment]
Hi, @zhoxing-ms, can you please help on reviewing this PR?

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py [hunk 1]
```diff
@@ -312,6 +312,8 @@ def ensure_container_insights_for_monitoring(
                 workspace_resource_id, "2015-11-01-preview"
             )
             location = resource.location
+            # location can have spaces for example 'East US' hence remove the spaces
+            location = location.replace(" ", "").lower()
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py:316]: Could you refer to this method for the processing of location names?

https://github.com/Azure/azure-cli/blob/b6b5cbf4613ae22b6116a93fcf17042adc563abb/src/azure-cli-core/azure/cli/core/commands/parameters.py#L95-L100

**Comment by zhoxing-ms** [comment]
@ganga1980 Since this PR is a bug fix, so it needs to be written into the history notes. Please check whether the PR title I modified meets your expectation

File: src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py (hunk 1 above)
Comment (ganga1980) [reply src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py:316]: thanks.

**Comment by ganga1980** [comment]
Hi, @zhoxing-ms,  can you please help on merging this PR?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**